### PR TITLE
test: add test on marathon and swarm hatchery

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report ut-engine#11345.0 ✘  
* Stage 1
  * Engine Test with DB ✘

**Comment by ovh-cds** [comment]
CDS Report it#11345.0 ✘  
* w-it
  * w-it ✘

**Comment by ovh-cds** [comment]
CDS Report ut-engine#11372.0 ✘  
* Stage 1
  * Engine Test with DB ✘

**Comment by ovh-cds** [comment]
CDS Report it#11372.0 ✘  
* w-it
  * w-it ✘

File: engine/hatchery/marathon/marathon.go [hunk 1]
```diff
@@ -366,55 +364,61 @@ func (h *HatcheryMarathon) SpawnWorker(ctx context.Context, spawnArgs hatchery.S
 
 	_, next = observability.Span(ctx, "waitDeployment")
 	wg := &sync.WaitGroup{}
-	var done bool
-	var successChan = make(chan bool, len(deployments))
+	var successChan = make(chan error, len(deployments))
```
Comment (bnjjj) [inline engine/hatchery/marathon/marathon.go]: maybe call this `errorChan` :p

File: engine/hatchery/swarm/swarm.go [hunk 2]
```diff
@@ -461,14 +472,14 @@ func (h *HatcherySwarm) CanSpawn(model *sdk.Model, jobID int64, requirements []s
...
 
 		//List all workers
-		ws, errWList := h.getWorkerContainers(dockerClient, cs, types.ContainerListOptions{})
+		ws, errWList := h.getWorkerContainers(cs, types.ContainerListOptions{})
 		if errWList != nil {
 			log.Error("hatchery> swarm> CanSpawn> Unable to list workers on %s: %s", dockerName, errWList)
 			continue
 		}
 
 		//Checking teh number of container on each docker engine
```
Comment (bnjjj) [inline engine/hatchery/swarm/swarm.go]:
```suggestion
		//Checking the number of container on each docker engine
```

**Comment by ovh-cds** [comment]
CDS Report it#11380.0 ✘  
* w-it
  * w-it ✘

**Comment by ovh-cds** [comment]
CDS Report ut-engine#11380.0 ✘  
* Stage 1
  * Engine Test with DB ✘

**Comment by ovh-cds** [comment]
CDS Report ut-engine#11389.0 ✘  
* Stage 1
  * Engine Test with DB ✘

**Comment by ovh-cds** [comment]
CDS Report build-engine-cli-tools#11389.0 ✘  
* Build minimal
  * cdsctl linux ✔ 
  * check SQL Files number ✔ 
  * Engine Build linux amd64 ✔ 
  * tools/debpacker ✘  
  * tools/os-ansible-inventory ✔ 
  * Worker linux amd64 / 386 ✔

**Comment by ovh-cds** [comment]
CDS Report ut-engine#11392.0 ✘  
* Stage 1
  * Engine Test with DB ✘

**Comment by ovh-cds** [comment]
CDS Report it#11392.0 ✘  
* w-it
  * w-it ✘

**Comment by ovh-cds** [comment]
CDS Report ut-engine#11396.0 ✘  
* Stage 1
  * Engine Test with DB ✘

**Comment by ovh-cds** [comment]
CDS Report it#11396.0 ✘  
* w-it
  * w-it ✘

File: engine/hatchery/marathon/marathon_test.go [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+	assert.Equal(t, "spawnMarathonDockerWorker> deployment for aaa timeout", err.Error())
+}
+
+func TestCanSpawn(t *testing.T) {
+	defer gock.Off()
+	h := InitMarathonMarathonTest(marathonJDD{
+		MaxWorker:    1,
+		MaxProvision: 1,
+	})
+	gock.New("http://mara.thon").Get("/v2/deployments").Reply(200).JSON([]*marathon.DeploymentID{})
```
Comment (bnjjj) [inline engine/hatchery/marathon/marathon_test.go]: http://mare-a.thon or http://flaque-a.truite ? :D

**Comment by ovh-cds** [comment]
CDS Report ut-engine#11485.0 ✘  
* Stage 1
  * Engine Test with DB ✘

**Comment by ovh-cds** [comment]
CDS Report cds-package#11485.1 ✘  
* build-package
  * Debian Engine Package ✘  
  * Docker Engine Package ✔ 
  * Package UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11499.0 ✘  
* Build
  * Build and Package All ✔ 
  * Engine Test with DB ✘  
  * UI Lint Translate ✔ 
  * UI Tests ✔

**Comment by ovh-cds** [comment]
CDS Report cds-package#11499.0 ✘  
* build-package
  * Debian Engine Package ✘  
  * Docker Engine Package ✔ 
  * Package UI ✔

**Comment by ovh-cds** [comment]
CDS Report ut-engine#11499.0 ✘  
* Stage 1
  * Engine Test with DB ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#11501.0 ✘  
* Build
  * Build and Package All ✔ 
  * Engine Test with DB ✔ 
  * UI Lint Translate ✔ 
  * UI Tests ✔ 
* Package
  * Docker ✔ 
  * Docs ✔ 
* Integration
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report cds-package#11501.0 ✘  
* build-package
  * Debian Engine Package ✘  
  * Docker Engine Package ✔ 
  * Package UI ✔

**Pull request merged**